edk2toollib/uefi/edk2/parsers/base_parser_test.py

```diff
@@ -39,6 +39,16 @@ def test_replace_macro_using_dollarsign(self):
         })
         line = "Hello $(name)!"
         self.assertEqual(parser.ReplaceVariables(line), "Hello sean!")
+    
+    def test_replace_macro_using_dollarsign_spaces(self):
+        parser = BaseParser("")
+        parser.SetInputVars({
+            "name": "sean"
+        })
+        line = "Hello $ (name)!"
+        self.assertEqual(parser.ReplaceVariables(line), "Hello sean!")
+        line = "Hello $( name )!"
+        self.assertEqual(parser.ReplaceVariables(line), "Hello sean!")
```

Comment:
Can you add a test for the negative case? e.g. missing $

Reply:
like `line = "Hello name"` or `line = "Hello $name"`?

Reply:
First one, then the other! :)